compiler/data/vertex-adaptor.h

```diff
@@ -7,170 +7,159 @@
   }
 
   template<Operation to>
-  VertexAdaptor<to> as() const {
-    if (!impl) {
+  VertexAdaptor<to> as() const noexcept {
+    if (!impl_) {
       return {};
     };
     auto res = try_as<to>();
-    kphp_assert_msg(res, fmt_format("Can't cast VertexAdaptor<{}>(real type {}) to VertexAdaptor<{}>",
-                                    OpInfo::op_str(Op), impl ? OpInfo::op_str(impl->type()) : "nullptr", OpInfo::op_str(to)));
+    kphp_assert_msg(res, fmt_format("Can't cast VertexAdaptor<{}>(real type {}) to VertexAdaptor<{}>", OpInfo::op_str(Op),
+                                    impl_ ? OpInfo::op_str(impl_->type()) : "nullptr", OpInfo::op_str(to)));
     return res;
   }
 
   template<Operation to>
-  VertexAdaptor<to> try_as() const {
-    if (!impl) {
+  VertexAdaptor<to> try_as() const noexcept {
+    if (!impl_) {
       return {};
     };
     static_assert(op_type_is_base_of(Op, to), "Strange downcast to not derived vertex");
     static_assert(Op != to || Op == meta_op_base, "Useless cast");
-    if (op_type_is_base_of(to, impl->type())) {
-      return VertexAdaptor<to>{static_cast<vertex_inner<to> *>(impl)};
+    if (op_type_is_base_of(to, impl_->type())) {
+      return VertexAdaptor<to>{static_cast<vertex_inner<to> *>(impl_)};
     }
 
     return {};
   }
 
-  static void init_properties(OpProperties *x) {
-    vertex_inner<Op>::init_properties(x);
+  static void init_properties(OpProperties *prop) {
+    vertex_inner<Op>::init_properties(prop);
   }
 
   template<typename... Args>
-  static VertexAdaptor<Op> create(Args &&... args) {
-    return VertexAdaptor<Op>(vertex_inner<Op>::create(std::forward<Args>(args)...));
+  static VertexAdaptor<Op> create(Args &&...args) noexcept {
+    return VertexAdaptor<Op>{vertex_inner<Op>::create(std::forward<Args>(args)...)};
   }
 
   template<typename... Args>
-  static VertexAdaptor<Op> create_vararg(Args &&... args) {
-    return VertexAdaptor<Op>(vertex_inner<Op>::create_vararg(std::forward<Args>(args)...));
+  static VertexAdaptor<Op> create_vararg(Args &&...args) noexcept {
+    return VertexAdaptor<Op>{vertex_inner<Op>::create_vararg(std::forward<Args>(args)...)};
   }
 
-  VertexAdaptor<Op> clone() const {
-    if (impl == nullptr) {
-      return {};
-    }
-    return VertexAdaptor<Op>(clone_vertex(*this).template as<Op>());
+  VertexAdaptor<Op> clone() const noexcept {
+    return impl_ ? VertexAdaptor<Op>(clone_vertex(*this).template as<Op>()) : VertexAdaptor<Op>{};
   }
 
-  template<Operation Op2>
-  friend class VertexAdaptor;
-
   bool operator==(const VertexAdaptor<Op> &other) const noexcept {
-    return impl == other.impl;
+    return impl_ == other.impl_;
   }
 
   bool operator!=(const VertexAdaptor<Op> &other) const noexcept {
-    return impl != other.impl;
+    return impl_ != other.impl_;
   }
 
-  VertexAdaptor &set_rl_type(RLValueType valueType) {
-    impl->rl_type = valueType;
+  VertexAdaptor &set_rl_type(RLValueType valueType) noexcept {
```

Comment:
It seems inconsistent, that here and below we write just `VertexAdaptor` (no template instantiation), whereas above we write `VertexAdaptor<Op>`

Reply:
done